### DISCUSSION TITLE
HUD discussion thread

### DISCUSSION THREAD

**Discussion by no-lex** [project-imprimis/imprimis#10] [Ideas]
Let's talk HUD design.

Imprimis is going to need a HUD design that is clear, concise, and good looking.

**Comment by byronsharman** [comment]
I personally think HUDs should get out of the way and show the most important information prominently. A good HUD design is one that no one notices.

**Comment by no-lex** [comment]
Some images of UIs in other FPS:

Quake Champions:
https://cdn.gamer-network.net/2017/usgamer/Quake-Champions-Beta-Shot-(14).jpg
![Quake-Champions-Beta-Shot-(14)](https://user-images.githubusercontent.com/23445929/103721944-c6d3a000-4f83-11eb-99f4-d41f63af827b.jpeg)

TF2:
https://wiki.teamfortress.com/wiki/Payload#/media/File:Cart_Headstart.jpg
![Cart_Headstart](https://user-images.githubusercontent.com/23445929/103722091-12864980-4f84-11eb-856a-8e2537a520d7.jpeg)

Fortnite:
https://i.pcmag.com/imagery/reviews/00XpAmYpxd6wvLkvIO672zY-5.1569471329.fit_scale.size_1969x1108.png
![00XpAmYpxd6wvLkvIO672zY-5](https://user-images.githubusercontent.com/23445929/103722686-588fdd00-4f85-11eb-8246-a5bc1a826009.png)

Valorant:
https://www.nme.com/wp-content/uploads/2020/06/Valorant-review-MG-credit-stacey-henley@2000x1270.jpg
![Valorant-review-MG-credit-stacey-henley@2000x1270](https://user-images.githubusercontent.com/23445929/103722467-dd2e2b80-4f84-11eb-8c36-e759a4ca2608.jpeg)

CS:GO
https://www.maketecheasier.com/assets/uploads/2019/07/csgo-server-featured3.jpg
![csgi](https://user-images.githubusercontent.com/23445929/103722271-714bc300-4f84-11eb-8783-a17e4a0e9656.jpeg)

"obvious" details:

* health almost always goes on the bottom left, and if not there it's at the bottom mid
* ammo/weapon info always goes on the bottom right
* minimap information is usually at the top left
* time & mode info is usually at the top center
* ghosts live in the top right, (generally nothing gets put there)

**Comment by Erarnitox** [comment]
I know i have already posted this in the discord, but just that no one misses it, ill write it here as well.
I quite enjoy the GUI of old battlefield and cod games, so i designed a simple GUI layout in that fashion:
![simple](https://cdn.discordapp.com/attachments/667227859047284757/796155568121184276/HudLayout.jpg)

However what was already mentioned is true: a good GUI is a GUI that isn't invasive,
so i also like the idea of not having a HUD at all and instead draw the most important information on the gun itself,
like the bullets left.
Health could be visualized through different means, like a bloody screen or wounds on the characters arms etc. that could be controlled through an alpha map. 

an example of using the weapon itself as additional space to draw information on would be halo:
![halo](https://www.wepc.com/wp-content/uploads/2020/04/Halo-2-anniversary-campaign-PC.png)

However, a lack of appealing visual feedback on the UI, like hit-markers or other kind of positive visual feedback for the player
make the game feel less arcade and possibly lower the "game play fun factor". 

So a mixture could be used, to on one had give the player enough information through a minimal UI
and let the player decide if he wants to be exposed to more eye candy or not. 

Quick note here: these are just my opinions that could very well be pron to personal bias/preference.

**Reply by no-lex** [reply]
Doing things like ammo on the weapon model itself is ***hard*** and the opportunity cost of doing so is probably not worth it from my point of view.

**Comment by byronsharman** [comment]
The Fortnite Chapter 2 HUD places the health at the bottom left.

![pqxlx1v3nq041](https://user-images.githubusercontent.com/39754510/103726226-a1e82880-4f95-11eb-9376-7ca36d441867.jpg)

**Reply by no-lex** [reply]
Fortnite has a thing for upper-right hand corner minimaps, I guess

**Comment by byronsharman** [comment]
Also, I only see horizontal health indicators.

**Reply by no-lex** [reply]
Yeah, it would probably be good to switch over, I think it has better symmetry with the horizontal weapon UI on the opposite side.

**Reply by no-lex** [reply]
Oh and technically TF2 has a vertical health indicator (albeit one that is a symmetrical cross)

**Reply by no-lex** [reply]
![nnxui](https://user-images.githubusercontent.com/23445929/103866851-48f2c000-507b-11eb-8a86-7e3911af177f.png)
Pushed a sideways health bar, probably not great but it's there to try.

**Reply by no-lex** [reply]
![newhud](https://user-images.githubusercontent.com/23445929/103957215-17233d00-50ff-11eb-9df7-62dd56dc07d9.png)
@nonoesimposible 's better healthbar UI added in #11
